src/Build/Evaluation/LazyItemEvaluator.cs

```diff
@@ -412,7 +425,7 @@ private LazyItemList GetItemList(string itemType)
 
         public IEnumerable<ItemData> GetAllItemsDeferred()
         {
-            return _itemLists.Values.SelectMany(itemList => itemList.GetItems(ImmutableHashSet<string>.Empty))
+            return _itemLists.Values.SelectMany(itemList => itemList.GetItemData(ImmutableHashSet<string>.Empty))
                                     .OrderBy(itemData => itemData.ElementOrder);
```

Comment:
This method produces *a lot* of allocations due to the sort - it looks like we only expect consume the sorting for conditioned properties. Does this have to be ordered? If so, can use underlying collection that preserves that?

Reply:
Tried to take it out and one test failed:

```
Microsoft.Build.UnitTests.Evaluation.Evaluator_Tests.AllEvaluatedItems [FAIL]
      Assert.Equal() Failure
                 (pos 0)
      Expected: j1
      Actual:   i3
                 (pos 0)
```

Also, even though there sadly isn't a test for this, `Project.Items` are also sorted, not just `Project.AllEvaluatedItems`.  We'd regress behaviour.
If we say that it's OK for `Project.Items` not to be sorted and potentially break people, it will make `Project.AllEvaluatedItems` be the only one returning sorted items. But that one is conditioned on `ShouldEvaluateForDesign`, which will be turned off soon by CPS and will start throwing.

Reply:
Projects.Items is backed by ItemDictionary and couldn't remember if that returned sorted order. I think I'm going to file this - there's a win here.